src/app/app.component.ts

```diff
@@ -1,24 +1,26 @@
-import { Actions, JsonSchema, UISchemaElement } from '@jsonforms/core';
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { Actions } from '@jsonforms/core';
+import { Component, Input, OnChanges, SimpleChanges, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
 
 import { JsonFormsAngularService } from '@jsonforms/angular';
 
 @Component({
-  selector: 'app-root',
+  selector: 'ng-jsonforms',
   template: `<jsonforms-outlet></jsonforms-outlet>`,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class AppComponent {
+export class AppComponent implements OnChanges{
+
+  @Input() uischema:string;
+  @Input() schema:string;
 
   constructor(private jsonformsService: JsonFormsAngularService, private cdr: ChangeDetectorRef) {
   }
 
-  setJsonFormsInput(
-    schema: JsonSchema,
-    uiSchema?: UISchemaElement,
-    data: any = {}
-  ) {
-    this.jsonformsService.updateCore(Actions.init(data, schema, uiSchema));
+  ngOnChanges(changes: SimpleChanges): void {
+    const schemaObject = JSON.parse(this.schema);
```

Comment:
can't it happen that you don't have a schema defined?

Reply:
It would be a valid use case for JSONForms only if the data is specified, right?

Reply:
I can add a `data` input and account for an undefined schema.